src/app/components/compact-list/directives/editable.directive.ts

```diff
@@ -0,0 +1,58 @@
+import {Directive, ElementRef, forwardRef} from "@angular/core";
+import {ControlValueAccessor, NG_VALUE_ACCESSOR, NG_VALIDATORS, FormControl} from "@angular/forms";
+
+@Directive({
+    selector: '[editable]',
```

Comment:
This still isn't an adequate selector for this directive because it needs `contenteditable` in order to work. If you stick it on any old div, for example, it won't work as expected. Make the selector `[contenteditable="true"][ct-editable]`  (or whatever name makes sense for this directive)